Added event types and their descriptions (#402)
typescript-cheatsheets/react: docs/basic/getting-started/forms-and-events.md

[View in the TypeScript Playground](https://www.typescriptlang.org/play/?jsx=2#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wFgAoCtCAOwGctoRlM4BeRYmAOgFc6kLABQBKClVoM4AMSbs4o9gD4FFOHAA8mJmrhFMbAN7aozJJgC+u2gGVeAIxDAYRoUgBcndDxsBPGjAAFkgwwGgAogBuSAEiynCGuupI3GBE0QEAIuYovAA2MKIA3Elw1PTwMChQAOYh8ilVtfUodHAwvmBIEKyN1XXwAGQJpckgKMB5noZwkSh5vB5wDFDANDVwFiXk6rtwYK10AO7QACbTs-OLnitrG1ulDzu75VJI45PyTQPc7xN53DmCyQRTgAHowe1Okg0ME0ABrOgAQlKr3gBzoxzOX36IVShxOUFOgKuIPBkI6XVhMMRKOe6ghcBCaG4rN0Fis5CUug0p2AkW59M0eRQ9iQeUFe3U4Q+U1GmjWYF4lWhbAARH9Jmq4DQUCAkOrNXltWDJbsNGCRWKJTywXyBTz7Wb1BoreLnbsAAoEs7ueUaRXKqFddUYrFE7W6-Whn0R8Eei1um3PC1Ox38hOBlUhtV0BxOGDaoGLdUAGQgGzWJrNqYzFAtJhAgpEQA)

Of course, if you're making any sort of significant form, [you should use Formik](https://jaredpalmer.com/formik) or [React Hook Form](https://react-hook-form.com/), which are written in TypeScript.

> **List of event types**
> | Event Type  | Description                                                                                                                   |

Reviewer: ```suggestion
### List of event types

| Event Type  | Description                                                                                                                   |
```

| TouchEvent       | Events that occur due to the user interacting with a touch device. Extends UIEvent.                                                                                                                           |
| TransitionEvent  |  CSS Transition. Not fully browser  supported. Extends UIEvent                                                                                                                             |
| UIEvent          |  Base Event for Mouse, Touch and Pointer events.                                                                                                                             |
| WheelEvent       | Scrolling on a mouse wheel or similar input device. (Note:  `wheel` event should not be confused with the `scroll` event)                                                                                                                              |
| SyntheticEvent   |  The base event for all above events. Should be used when unsure about event type                                                                                                                           |

> > **InputEvent ?**
> > You probably noticed that there is no InputEvent. This is because it is not supported by Typescript as the event itself has no fully browser support and may behave differently in different browsers.
> > It's suggested to use `KeyboardEvent` instead
> > Sources:
> >
> > - https://github.com/microsoft/TypeScript/issues/29441
> > - https://developer.mozilla.org/en-US/docs/Web/API/InputEvent
> > - https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event

Reviewer: ```suggestion
<details>
<summary>

**What about `InputEvent`?**

</summary>

You've probably noticed that there is no `InputEvent`. This is because it is not supported by Typescript as the event itself has no fully browser support and may behave differently in different browsers. You can use `KeyboardEvent` instead.

Sources:

 - https://github.com/microsoft/TypeScript/issues/29441
 - https://developer.mozilla.org/en-US/docs/Web/API/InputEvent
 - https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event

</details>
```
